47.md

```diff
@@ -135,6 +137,95 @@ The **client** should then store this connection and use it when the user wants
 nostr+walletconnect://b889ff5b1513b641e2a139f661a661364979c5beee91842f8f0ef42ab558e9d4?relay=wss%3A%2F%2Frelay.damus.io&secret=71a8c14c1407c113601079c4302dab36460f0ccd0ad506f1f2dc73b5100e4f3c
 ```
 
+## Client-Created Secret
+
+The client MAY generate its own secret and co-ordinate only public keys with the wallet service so that secrets are never exposed.
+
+### HTTP Confirmation Page
+
+A **wallet service** MAY host a publicly accessible confirmation page that the **client** may open (e.g. in a webview or popup window) and allow the **user** to confirm the creation of a new connection, with the following parameters:
+
+- `pubkey` Required. The corresponding pubkey of the secret generated by the **client**.
+
+The **wallet service** MAY ignore all the below optional parameters.
```

Comment:
I think it'd be good to reference an optional [NIP-68](https://github.com/nostr-protocol/nips/pull/1383) client ID here for some basic phishing protection.

Reply:
In the long term I think it's a good idea but I am unsure if it will be adopted in the short-mid term due to much more additional work on all sides (wallet service having to fetch the event and calculate WoT, client must handle a redirect uri, developer required to publish this event).

Should it be a NIP-19 `nevent` so that it also contains the relay? Do you have suggestions on the field name?